### PR TITLE
Add requirements.txt for easy venv setup

### PR DESCRIPTION
**Pull request by alexandrevicenzi**

### CODE REVIEW

**Comment by rohanpm** [comment]
Could you help me understand what's the purpose of this?  The deps are already in setup.py, does some tool not respect that?

If it makes sense to be done for this project, I guess it makes sense to be done for all of them and so should be added to some guidelines, along with the rationale.

**Comment by alexandrevicenzi** [comment]
@rohanpm what's the way to develop this project?

What I usually do is to run `pip install -r requirements.txt` to setup the dev environment, but if the correct way to develop is to install using setup.py this file is not necessary.

Also, setup.py could read requirements from this file.

**Comment by rohanpm** [comment]
The idea with the setup here was to follow de-facto standard Python conventions and not prescribe a specific development workflow.  So virtualenv & pip should work, ./setup.py develop should work, pipenv should work.

If there's a typical workflow which benefits from having the requirements.txt then it would make sense to add it.

Is ``pip install -r requirements.txt`` actually a good choice for a workflow though?  Since this is a library, generally you'd be building something on top of this.  But if you only install the dependencies of this library and not the library itself, I'm not sure how you can test that "something on top".  Do you manually set ``PYTHONPATH`` or something along those lines?

I'd typically have a virtualenv activated and then do ``pip install --editable .``, so the library and its dependencies are both available in the virtualenv, and the sources from the git clone are usable by other projects building on top of this one.

Let me know.  If what I said convinced you to use ``pip install --editable`` or equivalent, let's drop this; otherwise if you want to keep using ``pip install -r requirements.txt`` then let's merge this.

**Comment by alexandrevicenzi** [comment]
@rohanpm if you want to build something on top of this it makes more sense to install this library.

The idea of `requirements.txt` is to allow to work on this library. I usually do `pip install -r requirements.txt` and not `pip install --editable .`. I'm not against this way, but maybe we could have a couple of lines in the readme to show how to set up the env.

If we keep the `requirements.txt` we should at least use it in `setup.py`, otherwise, we need to update multiple files every time we change a package.

To sum it up I see two options:

1. Drop requirements.txt and state in the readme a "Development how to".
2. Keep requirements.txt and update setup.py to read this file.

If option 1 is chosen you can close this and I can send another patch to exemplify "Development how to".

If option 2 is chosen I can rework this patch and squash it.

As most of our projects do not really use requirements.txt maybe option 1 is better.

**Comment by rohanpm** [comment]
I thought there was some problem on older Python versions having setup.py read requirements.txt.  In fact there's no problem, I just hadn't set up MANIFEST.in correctly.

With the correct setup then, I don't really see any downside to having setup.py read requirements.txt, it seems to give best of both worlds and support all development workflows.  So let's do that.

I found a few other packaging issues, so I included the change to read from requirements.txt in my PR for those: https://github.com/release-engineering/python-fastpurge/pull/8
